com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/FullScreenCustomPassDrawer.cs

```diff
@@ -67,21 +76,147 @@ protected override void DoPassGUI(SerializedProperty customPass, Rect rect)
                         index = EditorGUI.IntPopup(rect, Styles.materialPassName, index, GetMaterialPassNames(mat), Enumerable.Range(0, mat.passCount).ToArray());
                         if (EditorGUI.EndChangeCheck())
                             m_MaterialPassName.stringValue = mat.GetPassName(index);
                     }
                     EditorGUI.EndProperty();
+                    rect.y += Styles.defaultLineSpace;
+
+                    GetStencilInfo(mat, out bool stencilEnabled, out int writeMask);
+
+                    if (stencilEnabled && targetDepthBuffer != CustomPass.TargetBuffer.Custom)
+                    {
+                        if (!m_ShowStencilInfoBox)
+                        {
+                            m_ShowStencilInfoBox = true;
+                            GUI.changed = true;
+                        }
+                        Rect helpBoxRect = rect;
+                        helpBoxRect.height = Styles.helpBoxHeight;
+                        helpBoxRect.xMin += Styles.indentPadding;
+                        EditorGUI.HelpBox(helpBoxRect, Styles.stencilHelpInfo, MessageType.Info);
+                        rect.y += Styles.helpBoxHeight;
+                    }
+                    else if (m_ShowStencilInfoBox)
+                    {
+                        m_ShowStencilInfoBox = false;
+                        GUI.changed = true; // Workaround to update the internal state of the ReorderableList and update the height of the element.
+                    }
+
+                    if (DoesWriteMaskContainsReservedBits(writeMask))
+                    {
+                        if (!m_ShowStencilWriteWarning)
+                        {
+                            m_ShowStencilWriteWarning = true;
+                            GUI.changed = true; // Workaround to update the internal state of the ReorderableList and update the height of the element.
+                        }
+                        Rect helpBoxRect = rect;
+                        helpBoxRect.height = Styles.helpBoxHeight;
+                        helpBoxRect.xMin += Styles.indentPadding;
+                        EditorGUI.HelpBox(helpBoxRect, Styles.stencilWriteOverReservedBits, MessageType.Warning);
+                        rect.y += Styles.helpBoxHeight;
+                    }
+                    else if (m_ShowStencilWriteWarning)
+                    {
+                        m_ShowStencilWriteWarning = false;
+                        GUI.changed = true; // Workaround to update the internal state of the ReorderableList and update the height of the element.
+                    }
                 }
             }
         }
 
+        bool DoesWriteMaskContainsReservedBits(int writeMask)
+        {
+            if (targetDepthBuffer == CustomPass.TargetBuffer.Custom)
+                return false;
+
+            return ((writeMask & (int)~(UserStencilUsage.UserBit0 | UserStencilUsage.UserBit1)) != 0);
```

Comment:
we should create a mask for all reserverd bit. Otherwise if in the future we change it we will need to edit all place like that. i.e do return ((writeMask & (int)~(UserStencilUsage.UserBitMask)

Reply:
done